backend/src/api/Filters/UserIdFilter.cs

```diff
@@ -0,0 +1,26 @@
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc;
+
+namespace api.Filters;
+
+public class UserIdFilter : Attribute, IAuthorizationFilter
```

Comment:
Sonarlint is complaining about not ending this class name with `Attribute`. In dotnet, is it a standard to use the postfix `Attribute` if we create an Attribute? 
Also, there is a violation of the [CA1018](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1018), should we always specify the `AttributeUsage`?